lms/services/jstor/factory.py

```diff
@@ -17,9 +17,7 @@ def service_factory(_context, request):
         # conveniently available unique user ID in a standardized format and
         # doesn't contain private info (email addresses etc.).
         headers={
-            "Tracking-User-ID": request.lti_user.h_user.username
-            if request.lti_user
-            else None,
+            "Tracking-User-ID": request.lti_user.h_user.username,
```

Comment:
`request.find_service(name="application_instance").get_current().settings`

above implies that there's already a lti_user, see:


https://github.com/hypothesis/lms/blob/0580df6f131383d49893fdcc2e6f584c71453fe5/lms/services/application_instance.py#L79


if there not `request.lti_user`, ApplicationInstanceNotFound will be raised and we'd never get to this conditional.

Reply:
Is there any chance that there will be a situation in future where we might have a "current" ApplicationInstance but not an LTI user?

Reply:
That situation might happen in the future but for now I rather make explicit that the user is AI is coming from lti user.

This PR goes into that direction: https://github.com/hypothesis/lms/pull/5334